src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTransformer.java

```diff
@@ -0,0 +1,340 @@
+    this.bundlingTransformer = new BundlingTransformer();
+  }
+
+  @Override
+  public List<String> getTemplateFileNames() {
+    return Arrays.asList(
+        XAPI_TEMPLATE_FILENAME, XSETTINGS_TEMPLATE_FILENAME, PACKAGE_INFO_TEMPLATE_FILENAME);
+  }
+
+  @Override
+  public List<ViewModel> transform(Model model, ApiConfig apiConfig) {
+    List<ViewModel> surfaceDocs = new ArrayList<>();
+    SurfaceNamer namer = new JavaSurfaceNamer();
+
+    List<ServiceDocView> serviceDocs = new ArrayList<>();
+    for (Interface service : new InterfaceView().getElementIterable(model)) {
+      SurfaceTransformerContext context =
+          SurfaceTransformerContext.create(service, apiConfig, createTypeTable(), namer);
+      StaticXApiView xapi = generateXApi(context);
+      surfaceDocs.add(xapi);
+
+      serviceDocs.add(xapi.doc());
+
+      context = SurfaceTransformerContext.create(service, apiConfig, createTypeTable(), namer);
+      StaticApiMethodView exampleApiMethod = getExampleApiMethod(xapi.apiMethods());
+      StaticXSettingsView xsettings = generateXSettings(context, exampleApiMethod);
+      surfaceDocs.add(xsettings);
+    }
+
+    PackageInfoView packageInfo =
+        generatePackageInfo(model, apiConfig, createTypeTable(), namer, serviceDocs);
+    surfaceDocs.add(packageInfo);
+
+    return surfaceDocs;
+  }
+
+  private ModelTypeTable createTypeTable() {
+    return new ModelTypeTable(new JavaTypeTable(), new JavaModelTypeNameConverter());
+  }
+
+  private StaticXApiView generateXApi(SurfaceTransformerContext context) {
+    addXApiImports(context);
+
+    List<StaticApiMethodView> methods = generateApiMethods(context);
+
+    StaticXApiView.Builder xapiClass = StaticXApiView.newBuilder();
+
+    ApiMethodView exampleApiMethod = getExampleApiMethod(methods);
+    xapiClass.doc(serviceTransformer.generateServiceDoc(context, exampleApiMethod));
+
+    xapiClass.templateFileName(XAPI_TEMPLATE_FILENAME);
+    xapiClass.packageName(context.getApiConfig().getPackageName());
+    String name = context.getNamer().getApiWrapperClassName(context.getInterface());
+    xapiClass.name(name);
+    xapiClass.settingsClassName(context.getNamer().getApiSettingsClassName(context.getInterface()));
+    xapiClass.apiCallableMembers(apiCallableTransformer.generateStaticApiCallables(context));
+    xapiClass.pathTemplates(pathTemplateTransformer.generatePathTemplates(context));
+    xapiClass.formatResourceFunctions(
+        pathTemplateTransformer.generateFormatResourceFunctions(context));
+    xapiClass.parseResourceFunctions(
+        pathTemplateTransformer.generateParseResourceFunctions(context));
+    xapiClass.apiMethods(methods);
+
+    // must be done as the last step to catch all imports
+    xapiClass.imports(context.getTypeTable().getImports());
+
+    String outputPath = pathMapper.getOutputPath(context.getInterface(), context.getApiConfig());
+    xapiClass.outputPath(outputPath + "/" + name + ".java");
```

Comment:
Maybe this could use a path builder of some sort, rather than hard-coded forward slash separator?